src/pudl/transform/eia860.py

```diff
@@ -748,3 +748,187 @@ def clean_boilers_eia860(
+    )
+    # Add a emission_control_id_pudl as a primary key. This is not unique over years.
+    # We could maybe try and do this, but not doing it now.
+    emce_df["emission_control_id_pudl"] = (
+        emce_df.groupby(["report_year", "plant_id_eia"]).cumcount() + 1
+    )
+    # Fix outlier value in emission_control_equipment_cost. We know this is an
+    # outlier because it is the highest value reported in the dataset and
+    # the other years from the same plant show that it likely contains three
+    # extra zeros. We use the primary keys to spot fix the value.
+    outlier_primary_keys = (
+        (emce_df["report_year"] == 2017)
+        & (emce_df["plant_id_eia"] == 57794)
+        & (emce_df["emission_control_equipment_cost"] == 3200000)
+    )
+
+    if len(emce_df[outlier_primary_keys]) > 2:
+        raise AssertionError("Only expecting two spot fixed values here")
+
+    emce_df.loc[
+        outlier_primary_keys,
+        "emission_control_equipment_cost",
+    ] = 3200
+
+    return emce_df
+
+
+@asset
+def clean_boiler_emissions_control_equipment_assn_eia860(
+    raw_boiler_so2_eia860: pd.DataFrame,
+    raw_boiler_mercury_eia860: pd.DataFrame,
+    raw_boiler_nox_eia860: pd.DataFrame,
+    raw_boiler_particulate_eia860: pd.DataFrame,
+) -> pd.DataFrame:
+    """Pull and transform the emissions control <> boiler ID link tables.
+
+    Args:
+        raw_boiler_so2_eia860: Raw EIA 860 boiler to SO2 emission control equipment
+            association table.
+        raw_boiler_mercury_eia860: Raw EIA 860 boiler to mercury emission control
+            equipment association table.
+        raw_boiler_nox_eia860: Raw EIA 860 boiler to nox emission control equipment
+            association table.
+        raw_boiler_particulate_eia860: Raw EIA 860 boiler to particulate emission
+            control equipment association table.
+        raw_boiler_cooling_eia860: Raw EIA 860 boiler to cooling equipment association
+            table.
+        raw_boiler_stack_flue_eia860: Raw EIA 860 boiler to stack flue equipment
+            association table.
+
+    Returns:
+        pd.DataFrame: A combination of all the emission control equipment association
+            tables.
+    """
+    raw_tables = [
+        raw_boiler_so2_eia860,
+        raw_boiler_mercury_eia860,
+        raw_boiler_nox_eia860,
+        raw_boiler_particulate_eia860,
+    ]
+
+    bece_df = pd.DataFrame({})
+
+    for table in raw_tables:
+        # There are some utilities that report the same emissions control equipment
+        table = table.drop_duplicates(
+            subset=[
+                x
+                for x in table.columns
+                if x not in ["utility_id_eia", "utility_name_eia"]
+            ]
+        )
+        value_col = [col for col in table if "control_id_eia" in col]
+        id_cols = [col for col in table if "control_id_eia" not in col]
+        assert len(value_col) == 1
```

Comment:
Could you add a little comment explaining this check? My understanding is that each table should have exactly one column with `control_id_eia` in it's name somewhere, but I find it helpful to have explicit documentation of these inline checks